---
title: "Assignment 3"
author: "Vignesh Narayanaswamy"
date: "February 11, 2020"
output: html_document
---

```{r}
setwd("C:/Users/arcaroadmin/Dropbox/2020 Bio697GA/Assignment 3")
SNPs <- read.table("23andMe_complete.txt", header = TRUE, sep = "\t")
library(ggplot2)
```

**Exercise 1**
*Using ggplot make a make a bar graph of the total SNP counts for each chromosome*
```{r}
ggplot(SNPs, aes(x=chromosome)) + geom_bar()
```

**Exercise 2**
*Order the chromosomes according to number by converting chromosomes from a factor to a order factor as in the example above. Then replot the bar graph*
```{r}
SNPs$chromosome = ordered(SNPs$chromosome, levels=c(seq(1,22), "X", "Y", "MT"))
ggplot(SNPs, aes(x=chromosome)) + geom_bar()
```

**Exercise 3**
*Show the contribution of each genotype to the chromosome count using a stacked bar graph (with the fill = genotype)*
```{r}
ggplot(SNPs, aes(x=chromosome, fill = genotype)) + geom_bar(color = "black")
```

**Exercise 4**
*Make each set of stacked bars the same height to easier to compare proportions across groups.*
```{r}
ggplot(SNPs, aes(x=chromosome, fill = genotype)) + geom_bar(position = "fill", color = "black")
```

**Exercise 5**
*Now place genotypes directly beside one another for each chromosome to compare individual values.*
```{r}
ggplot(SNPs, aes(x=genotype, fill = chromosome)) + geom_bar(position = "fill", color = "black")
```


**Exercise 6**
*The above graph is pretty hard to read. Try using facet_wrap with the genotype*
```{r}
ggplot(SNPs, aes(x=genotype, fill = chromosome)) + geom_bar(position = "fill", color = "black") + facet_wrap(~chromosome)
```

